xilem/src/view/canvas.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use masonry::widgets;
+use vello::Scene;
+use vello::kurbo::Size;
+
+use crate::core::{Arg, MessageCtx, MessageResult, Mut, View, ViewArgument, ViewMarker};
+use crate::{Pod, ViewCtx};
+
+/// # Example
+///
+/// ```
+/// use xilem::masonry::kurbo::{Rect, Size, Affine};
+/// use xilem::masonry::peniko::Fill;
+/// use xilem::masonry::vello::Scene;
+/// use xilem::view::canvas;
+/// use std::sync::Arc;
+/// # use xilem::WidgetView;
+///
+/// # fn fill_canvas() -> impl WidgetView<()> + use<> {
+/// let my_canvas = canvas(Arc::new(|scene: &mut Scene, size: Size| {
+///     // Drawing a simple rectangle that fills the canvas.
+///     scene.fill(
+///         Fill::NonZero,
+///         Affine::IDENTITY,
+///         xilem::palette::css::AQUA,
+///         None,
+///         &Rect::new(0.0, 0.0, size.width, size.height),
+///     );
+/// }));
+/// # my_canvas
+/// # }
+/// ```
+pub fn canvas(draw: Arc<dyn Fn(&mut Scene, Size) + Send + Sync + 'static>) -> Canvas {
```

Comment:
Btw. I think we should probably specify in the doc-comment, that this `Arc<dyn Fn>` should be ideally saved within the app state, otherwise it always results in a rerender. Ideally show how this is done better in the doc-comment example